File: backend/persistend/enums.py
from enum import Enum


# -----------------------------------------------------------------------------
# РОЛИ НА ХАКАТОНЕ
# -----------------------------------------------------------------------------
class RoleType(str, Enum):
    DevOps = "DevOps"
    GameDev = "GameDev"
    MobileDev = "MobileDev"
    ProductManager = "Product manager"
    DS = "DS"
    ML = "ML"
    Fullstack = "Fullstack"
    Backend = "Backend"
    Frontend = "Frontend"
    Designer = "Designer"
    Analytics = "Analytics"
    QA = "QA"


# -----------------------------------------------------------------------------
# СТАТУС АНКЕТЫ (application_status)
# -----------------------------------------------------------------------------
class ApplicationStatus(str, Enum):
    draft = "draft"
    published = "published"
    hidden = "hidden"


# -----------------------------------------------------------------------------
# РЕЖИМ И СТАТУС ХАКАТОНА
# -----------------------------------------------------------------------------
class HackathonMode(str, Enum):
    online = "online"
    offline = "offline"
    hybrid = "hybrid"


class HackathonStatus(str, Enum):
    open = "open"
    closed = "closed"


# -----------------------------------------------------------------------------
# СТАТУС КОМАНДЫ (team_status)
# -----------------------------------------------------------------------------
class TeamStatus(str, Enum):
    forming = "forming"
    ready = "ready"


# -----------------------------------------------------------------------------
# СТАТУС ВАКАНСИИ (vacancy_status)
# -----------------------------------------------------------------------------
class VacancyStatus(str, Enum):
    open = "open"
    closed = "closed"


# -----------------------------------------------------------------------------
# СТАТУС ИНВАЙТА (invite_status)
# -----------------------------------------------------------------------------
class InviteStatus(str, Enum):
    pending = "pending"
    accepted = "accepted"
    rejected = "rejected"
    expired = "expired"


# -----------------------------------------------------------------------------
# СТАТУС ОТКЛИКА (response_status)
# -----------------------------------------------------------------------------
class ResponseStatus(str, Enum):
    pending = "pending"
    accepted = "accepted"
    rejected = "rejected"
    withdrawn = "withdrawn"


# -----------------------------------------------------------------------------
# ДОСТИЖЕНИЯ (achiev_place)
# -----------------------------------------------------------------------------
class AchievementPlace(str, Enum):
    firstPlace = "firstPlace"
    secondPlace = "secondPlace"
    thirdPlace = "thirdPlace"
    finalyst = "finalyst"
    participant = "participant"


# -----------------------------------------------------------------------------
# ТИПЫ НОТИФИКАЦИЙ (notif_type)
# -----------------------------------------------------------------------------
class NotifType(str, Enum):
    info = "info"
    warning = "warning"

    response_accept = "response:accept"
    response_decline = "response:decline"

    invitation_new = "invitation:new"
    invitation_reject = "invitation:reject"
    invitation_accept = "invitation:accept"

    team_entry = "team:entry"
    team_leave = "team:leave"
    team_kicked = "team:kicked"
    
class InviteStatus(str, Enum):
    pending  = "pending"
    accepted = "accepted"
    rejected = "rejected"
    expired  = "expired"


class ResponseStatus(str, Enum):
    pending   = "pending"
    accepted  = "accepted"
    rejected  = "rejected"
    withdrawn = "withdrawn"

File: frontend/src/shared/types/enums.ts
// допустимые места
export const ACHIEVEMENT_PLACES = [
  "participant",
  "finalyst",
  "thirdPlace",
  "secondPlace",
  "firstPlace",
] as const;

export type AchievementPlace = (typeof ACHIEVEMENT_PLACES)[number];

// роли
export const ACHIEVEMENT_ROLES = [
  "Backend",
  "Frontend",
  "Fullstack",
  "Data",
  "Product",
  "Designer",
] as const;

export type AchievementRole = (typeof ACHIEVEMENT_ROLES)[number];


export const placeOptions: { label: string; value: AchievementPlace }[] = [
  { label: "1 место", value: "firstPlace" },
  { label: "2 место", value: "secondPlace" },
  { label: "3 место", value: "thirdPlace" },
  { label: "Участник", value: "participant" },
];

export const roleOptions: { label: string; value: AchievementRole }[] = [
  { label: "Backend", value: "Backend" },
  { label: "Frontend", value: "Frontend" },
  { label: "Fullstack", value: "Fullstack" },
  { label: "Data / ML", value: "Data" },
  { label: "Product", value: "Product" },
  { label: "Designer", value: "Designer" },
];